packages/core/admin/ee/admin/hooks/useLicenseLimitNotification/tests/index.test.js

```diff
@@ -0,0 +1,166 @@
+import { renderHook } from '@testing-library/react-hooks';
+import useLicenseLimitNotification from '..';
+import useLicenseLimits from '../../useLicenseLimits';
+
+const baseLicenseInfo = {
+  enforcementUserCount: 5,
+  currentActiveUserCount: 5,
+  permittedSeats: 5,
+  shouldNotify: true,
+  shouldStopCreate: true,
+  licenseLimitStatus: 'AT_LIMIT',
+  isHostedOnStrapiCloud: false,
+  licenseType: 'gold',
+};
+
+jest.mock('react-intl', () => {
+  return {
+    useIntl: jest.fn(() => ({
+      formatMessage: jest.fn((options) => options.defaultMessage),
+    })),
+  };
+});
```

Comment:
You don't have to mock this - by default react-intl will return the defaultMessage.

It would be better to wrap the in an IntlProvider. Example: https://github.com/strapi/strapi/blob/main/packages/core/upload/admin/src/hooks/tests/useBulkMove.test.js

This way you could also create a single setup function which renders the hook in context - makes the tests easier to change in the future.

Reply:
I don't think I agree on this one though as wrapping it in the `IntlProvider` will not do anything for the actual hook testing, but will also introduce testing of `react-intl`. The test will perform as expected without it

Reply:
It depends how you look at it: I always try to avoid mocking as much as possible because how we handle it (local for each test) is a pain when you need to do a large-scale refactoring one day. `IntlProvider` works the way it does and for the tests you'd use it like: `<IntlProvider messages={{}} locale="en" />` which is basically mocking without mocking.

If you have strong feelings about your approach feel free to keep it though.